### ISSUE TITLE
Show Git version

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15098]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

N/A

### Is your feature request related to a problem?

`rubocop -V` shows only release version which is quite granular, so if user is running (older) Git version, it may be unclear to devs whether the report is an actual regression / incomplete prior fix, or just a previous version.

### Describe the solution you'd like

Provide devs with precise information about which Git version user is running, if any.

### Code examples

```ruby
diff --git a/lib/rubocop/version.rb b/lib/rubocop/version.rb
index 5553fbc03..e103f370c 100644
--- a/lib/rubocop/version.rb
+++ b/lib/rubocop/version.rb
@@ -28,3 +28,3 @@ module RuboCop
         target_ruby_version = target_ruby_version(env)
-        verbose_version = format(MSG, version: STRING,
+        verbose_version = format(MSG, version: git_version || STRING,
                                       parser_version: parser_version(target_ruby_version),
@@ -158,2 +158,7 @@ module RuboCop
     end
+
+    def self.git_version
+      path = File.dirname(File.dirname(File.realpath($PROGRAM_NAME)))
+      `git -C "#{path}" log -1 --format="%h %cd%d"`.strip if File.exist?("#{path}/.git")
+    end
   end
```

### Describe alternatives you've considered

Users having to remember to point this out manually, which is of course unreliable

### Additional context

Patch is illustrative and without possible necessary tests, but assume solution is something along these lines.

Also, may need fallback for if git executable is N/A (albeit quite unlikely if user is running a Git Rubocop)

**Comment by koic** [comment]
Thank you for the suggestion. Typical users run the stable release. The proposed feature would require checking for the git executable, detecting the .git directory, and handling  subprocess errors. The edge-case handling outweighs the benefit.

**Comment by forthrin** [comment]
It's your bed :) Good to hear version snafus are not an issue, like certain other popular tools.
